demo/admin/src/products/ManufacturerForm.tsx

```diff
@@ -235,19 +235,14 @@ export function ManufacturerForm({ id }: FormProps) {
                                 label={<FormattedMessage id="manufacturer.address.country" defaultMessage="Address Country" />}
                             />
                             <Divider sx={{ marginBottom: 5 }} />
-                            <Field
+                            <SwitchField
                                 fullWidth
                                 name="useAlternativeAddress"
-                                type="checkbox"
-                                label={<FormattedMessage id="manufacturer.address.useAlternativeAddress" defaultMessage="Use alternative address" />}
-                            >
-                                {(props) => (
-                                    <FormControlLabel
-                                        control={<FinalFormSwitch {...props} />}
-                                        label={props.input.checked ? <FormattedMessage {...messages.yes} /> : <FormattedMessage {...messages.no} />}
-                                    />
-                                )}
-                            </Field>
+                                fieldLabel={
+                                    <FormattedMessage id="manufacturer.address.useAlternativeAddress" defaultMessage="Use alternative address" />
+                                }
+                                label={values.useAlternativeAddress ? <FormattedMessage {...messages.yes} /> : <FormattedMessage {...messages.no} />}
```

Comment:
It's odd that we need to subscribe to values to change the label here. Should we add support for passing a function `(checked: boolean) => ReactNode` to SwitchField?

Reply:
Yes that would probably make sense

Reply:
I added support for this with 998d0cb05a282548fac3679f0d814883b758eda7